tools/fetch-sponsors.js

```diff
@@ -387,6 +391,59 @@ async function fetchGitHubSponsors() {
 	};
 }
 
+/**
+ * Fetches thanks.dev data using the REST API.
+ * @returns {Promise<{sponsors: Array}>} An promise that resolves to an object with one property:
+ *  - `sponsors` (Array): List of sponsors
+ */
+async function fetchThanksDevData() {
+	const endpoint = "https://api.thanks.dev/v1/entity/gh/eslint/donors";
+
+	const { body: result } = await request(endpoint, {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	});
+
+	const payload = await result.json();
+
+	if (process.env.DEBUG) {
+		await fs.writeFile(
+			"./debug-thanksdev-raw-response.json",
+			JSON.stringify(payload, null, 4),
+			{ encoding: "utf8" },
+		);
+	}
+
+	const sponsors = Object.values(payload.donors)
+		.filter(({ payments }) => {
+			const lastPayment = payments.at(-1);
+
+			const now = new Date();
+			const isCurrentYear =
+				now.getFullYear().toString() === lastPayment.month.slice(0, 4);
+			const isCurrentMonth =
+				(now.getMonth() + 1).toString().padStart(2, "0") ===
+				lastPayment.month.slice(5, 7);
+
+			return isCurrentMonth && isCurrentYear;
+		})
+		.map(({ name, login, avatar, url, payments }) => {
+			const { amount } = payments.at(-1);
+			return {
+				name: name ?? login,
+				url: fixUrl(url),
+				image: avatar,
+				monthlyDonation: Number(amount),
```

Comment:
We can also calculate `totalDonations` by adding up the payments.

```js
totalDonations: payments.reduce((total, payment) => total + payment.amount, 0);
```

Reply:
Isnt this only for the month? Or do we want to show the total the user has paid?

Reply:
It should be for the month, I suppose. We won’t be able to show it to the user since we’re only checking their last payment.

Reply:
The sponsors from GitHub and Open Collective all include `totalDonations` along with `monthlyDonation`. I think we should do the same here.

Reply:
done added total donations